docs/docs/integrating-with-other-libraries.md

```diff
@@ -0,0 +1,270 @@
+
+In the following code, `ListComponent` renders a collection with `ItemComponent` responsible for rendering the individual models.
+
+```js
+class ItemComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.rerender = () => this.forceUpdate();
+  }
+
+  componentDidMount() {
+    this.props.model.on('change', this.rerender);
+  }
+
+  componentWillUnmount() {
+    this.props.model.off('change', this.rerender);
+  }
+
+  render() {
+    return <li>{this.props.model.get('text')}</li>;
+  }
+}
+
+class ListComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.rerender = () => this.forceUpdate();
+  }
+
+  componentDidMount() {
+    this.props.collection.on('add', 'remove', this.rerender);
+  }
+
+  componentWillUnmount() {
+    this.props.collection.off('add', 'remove', this.rerender);
+  }
+
+  render() {
+    return (
+      <ul>
+        {this.props.collection.map(model => (
+          <ItemComponent key={model.cid} model={model} />
+        ))}
+      </ul>
+    )
+  }
+}
+```
+
+[Try it on CodePen.](http://codepen.io/wacii/pen/oBdgoM?editors=0010)
+
+### Extracting data from Backbone models
+
+Alternatively, whenever a model changes, you can extract its attributes as plain data. The following is [a higher-order component (HOC)](/react/docs/higher-order-components.html) that extracts all attributes of a Backbone model into state, passing the data to the wrapped component.
+
+This way, only the HOC needs to know about Backbone model internals, and the components concerned with presenting data can focus on that.
+
+```js
+function backboneModelAdapter(Component) {
+  return class extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = { ...props.model.attributes };
+      this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(model) {
+      this.setState(model.changedAttributes());
+    }
+
+    componentDidMount() {
+      this.props.model.on('change', this.handleChange);
+    }
+
+    componentWillUnmount() {
+      this.props.model.off('change', this.handleChange);
+    }
+
+    render() {
+      const { model, ...otherProps } = this.props;
+      return <Component {...otherProps} {...this.state} />;
+    }
+  }
+}
+```
+
+A copy is made of the model's attributes to form the initial state. Every time there is a change event, just the changed attributes are updated.
+
+To demonstrate its use, we will use a basic input wrapper. The component will render the `'text'` attribute of the provided model, calling an update function whenever the input is changed, presumably to update the model.
+
+```js
+function Input({ text, handleChange }) {
+  return <input value={text} onChange={handleChange} />;
+}
+
+function BackboneModelAdapterDemo() {
+  const model = new Backbone.Model({ text: 'Sam' });
+  const handleChange = event => model.set('text', event.target.value);
+  const WrappedInput = backboneModelAdapter(Input);
```

Comment:
Don't do this inside the component. You're creating a new copy each render, destroying the state and DOM.